src/saving/SaveHelper.cs

```diff
@@ -131,11 +131,96 @@ private static Save CreateSaveObject(MainGameState gameState, SaveInformation.Sa
         };
     }
 
+    private static string BuildQuickSaveFilename(int quickSaveId)
+    {
+        return $"quick_save_{quickSaveId}.{Constants.SAVE_EXTENSION}";
+    }
+
+    /// <summary>
+    ///   Returns the next quick save id to be used
+    /// </summary>
+    private static int GetNextQuickSaveId()
+    {
+        var files = GetSaves();
+        var enumerable = files as string[] ?? files.ToArray();
+        for (var i = 0; i < Settings.Instance.MaxQuickSavesBeforeOverriding; i++)
+        {
+            if (!enumerable.Contains(BuildQuickSaveFilename(i)))
+            {
+                // Not all 5 quick saves have been used
+                return i;
+            }
+        }
+
+        // All 5 quick saves have been used
+        var filename = GetLastSaveGame(true);
+
+        // Get the id out of the quick save file name.
+        filename = filename.Substr("quick_save_".Length,
+            filename.Length - Constants.SAVE_EXTENSION.Length - "quick_save_".Length - 1);
+
+        var id = int.Parse(filename, System.Globalization.CultureInfo.InvariantCulture);
+
+        // Get the next id. Overflow if MaxQuickSavesBeforeOverriding is reached.
+        return (id + 1) % (Settings.Instance.MaxQuickSavesBeforeOverriding + 1);
```

Comment:
I still don't like this part. What I'd prefer is if you did something like:
- get full paths from files with a map, and get oldest modified file, and return that filename